spacetypes.h

```diff
@@ -0,0 +1,47 @@
+#ifndef __SPACETYPES_H
```

Comment:
Note: I'm including this file in the review, but it should be a standard header. These are the structs each team told me they were using and needed to be logged.